cardano-api/src/Cardano/Api/ScriptData.hs

```diff
@@ -77,14 +89,35 @@ import           Cardano.Api.Keys.Shelley
 import           Cardano.Api.SerialiseCBOR
 import           Cardano.Api.SerialiseJSON
 import           Cardano.Api.SerialiseRaw
-import qualified Cardano.Binary as CBOR
-
 import           Cardano.Api.SerialiseUsing
 import           Cardano.Api.TxMetadata (pBytes, pSigned, parseAll)
-import           Codec.Serialise.Class (Serialise (..))
+
+-- Original script data bytes
+data HashableScriptData
+  = HashableScriptData
+      !BS.ByteString -- ^ Original 'ScriptData' bytes
+      !ScriptData
+      deriving (Eq, Show)
+
+instance HasTypeProxy HashableScriptData where
+    data AsType HashableScriptData = AsHashableScriptData
+    proxyToAsType _ = AsHashableScriptData
+
+instance SerialiseAsCBOR HashableScriptData where
+    serialiseToCBOR (HashableScriptData _ sd) = CBOR.serialize' sd
```

Comment:
```suggestion
    serialiseToCBOR (HashableScriptData bs _) = bs
```
Maybe I am confused, but is the point of `HashableScriptData` to make sure that if you ever hash a value of `HashableScriptData` that you hash the original bytes? If so, wouldn't you want to make sure that you always retain the original serialization?

I think it would also make sense to hide the first field (`!BS.ByteString`) with a smart constructor so that `HashableScriptData` can't be constructed incorrectly.

Reply:
> Maybe I am confused, but is the point of HashableScriptData to make sure that if you ever hash a value of HashableScriptData that you hash the original bytes? If so, wouldn't you want to make sure that you always retain the original serialization?

Nop you're right, this was a mistake.